File: 1erdepartamental/python/17.merge_sort/main.py
"""
Programa: Merge Sort
Demuestra algoritmo de ordenamiento merge sort
"""

from core.ordenador import OrdenadorMerge

if __name__ == "__main__":
    numeros = [64, 34, 25, 12, 22, 11, 90]

    print(f"Array original: {numeros}")

    ordenador = OrdenadorMerge()
    ordenador.ordenar(numeros)

    print(f"Array ordenado: {numeros}")
    print(f"Fusiones: {ordenador.fusiones}")

File: 1erdepartamental/c++/17.merge_sort/main.cpp
#include <iostream>
#include "operations/OrdenadorMerge.h"

int main() {
    std::vector<int> numeros = {64, 34, 25, 12, 22, 11, 90};
    
    std::cout << "Array original: ";
    for (int n : numeros) std::cout << n << " ";
    std::cout << std::endl << std::endl;
    
    OrdenadorMerge<int> ordenador;
    ordenador.ordenar(numeros);
    
    std::cout << "Array ordenado: ";
    for (int n : numeros) std::cout << n << " ";
    std::cout << std::endl << std::endl;
    
    std::cout << "Fusiones: " << ordenador.getFusiones() << std::endl;
    
    return 0;
}
